### PR TITLE
Debug error: Inject path where deserializations fails

### PR DESCRIPTION
**Pull request by TilakMaddy**

### CODE REVIEW

File: aderyn_driver/src/process_auto.rs [hunk 1]
```diff
@@ -162,7 +162,16 @@ fn absorb_ast_content_into_context(
...
+    let source_unit_deserializer = &mut serde_json::Deserializer::from_str(ast_content);
+    let result: Result<SourceUnit, _> = serde_path_to_error::deserialize(source_unit_deserializer);
+    let mut source_unit: SourceUnit = match result {
+        Ok(source_unit) => source_unit,
+        Err(err) => {
+            let path = err.path().to_string();
+            println!("Faieled to deserialize at path {:?}", path);
+            std::process::exit(-1);
+        }
+    };
```
Comment (TilakMaddy) [inline aderyn_driver/src/process_auto.rs:174]: 2nd thoughts .. before merging. Deserializing this way could be very slow and I think it should only be used in tests.

Comment (TilakMaddy) [reply aderyn_driver/src/process_auto.rs:174]: For now let us only use this branch for debugging purposes. I think once the AST is complete, we wouldn't need to worry too much

**Comment by alexroan** [comment]
@TilakMaddy error trace:

```
thread 'main' panicked at /Users/alexroan/Documents/Cyfrin/aderyn/aderyn_core/src/ast/source_units.rs:47:46:
called `Result::unwrap()` on an `Err` value: Error("missing field `referencedDeclaration`", line: 0, column: 0)
stack backtrace:
   0: _rust_begin_unwind
   1: core::panicking::panic_fmt
   2: core::result::unwrap_failed
   3: <aderyn_core::ast::source_units::SourceUnitNode as serde::de::Deserialize>::deserialize
   4: <serde_path_to_error::de::TrackedSeed<X> as serde::de::DeserializeSeed>::deserialize
   5: <serde_json::de::SeqAccess<R> as serde::de::SeqAccess>::next_element_seed
   6: <serde::de::impls::<impl serde::de::Deserialize for alloc::vec::Vec<T>>::deserialize::VecVisitor<T> as serde::de::Visitor>::visit_seq
   7: <&mut serde_json::de::Deserializer<R> as serde::de::Deserializer>::deserialize_seq
   8: <serde_path_to_error::de::Deserializer<D> as serde::de::Deserializer>::deserialize_seq
   9: <serde_path_to_error::de::TrackedSeed<X> as serde::de::DeserializeSeed>::deserialize
  10: serde_path_to_error::de::<impl serde::de::Visitor for serde_path_to_error::wrap::Wrap<X>>::visit_map
  11: <&mut serde_json::de::Deserializer<R> as serde::de::Deserializer>::deserialize_struct
  12: <serde_path_to_error::de::Deserializer<D> as serde::de::Deserializer>::deserialize_struct
  13: serde_path_to_error::de::deserialize
  14: aderyn_driver::process_auto::absorb_ast_content_into_context
  15: aderyn_driver::process_auto::with_project_root_at::{{closure}}
  16: rayon::iter::plumbing::Folder::consume_iter
  17: rayon::iter::plumbing::bridge_producer_consumer::helper
  18: <rayon::vec::IntoIter<T> as rayon::iter::IndexedParallelIterator>::with_producer
  19: rayon::iter::from_par_iter::collect_extended
  20: aderyn_driver::process_auto::with_project_root_at
  21: aderyn_driver::driver::drive_with
  22: aderyn_driver::driver::drive
  23: aderyn::main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

**Comment by alexroan** [comment]
It might be the BTreeMap you added in the ContractDefinition AST?

**Comment by TilakMaddy** [comment]
Hmm ..but this is not how it's supposed to look like - 

See this example - https://github.com/dtolnay/path-to-error 
printed error path contains the actual field names

**Comment by TilakMaddy** [comment]
Could you retry on this branch?

I'm really curious which AST node is causing the issue :P

**Pull request closed without merging**